rotkehlchen/assets/asset.py

```diff
@@ -834,6 +834,19 @@ def resolve_to_fiat_asset(self) -> 'FiatAsset':
             expected_type=FiatAsset,
         )
 
+    def symbol_or_name(self) -> str:
+        """
+        If it is an asset with symbol, returns symbol. If it's not, returns name.
+        May raise:
+        - UnknownAsset if identifier is not in the db
+        """
+        try:
+            with_symbol = self.resolve_to_asset_with_symbol()
+            return with_symbol.symbol
+        except WrongAssetType:
+            with_name = self.resolve_to_asset_with_name_and_type()
```

Comment:
I like the duck typing/error handling here. But can't thisalso raise `WrongAssetType` too if the `self` asset can't be resolved to `AssetWithNameAndType`? In which case you would need to just add it to the "May raise" part of the docstring.

I think it's fair to say this function can raise both ... or if you prefer you can catch it again and raise `UnknownAsset` ?

Btw thinking about it more ... am I not wrong here? Everything should be resolvable to `AssetWithNameAndType`? Right? If not that means it does not exist in the DB, right?

Since **every** asset has both a name and a type.

Reply:
Exactly. Everything is resolvable to `AssetWithNameAndType` since `asset_type` and `name` are in the same table as `identifier`.

Reply:
thanks makes sense.